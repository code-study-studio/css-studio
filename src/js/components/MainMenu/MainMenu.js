// @flow

import React from 'react';
import { NavLink } from 'react-router-dom';
import './MainMenu.scss';

type Props = {
  // onLinkSelect: (id: number) => void,
}

const selectedStyle = {
  backgroundColor: 'white',
  color: 'slategray',
};

const MainMenu = (props: Props) => (
  <nav className='menu'>
    <NavLink activeStyle={selectedStyle} to='/'>
      [Home]
    </NavLink>
  </nav>
);

export default MainMenu;