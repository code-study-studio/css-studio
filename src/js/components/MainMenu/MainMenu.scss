@import '../../../styles/mixins';
@import '../../../styles/vars';


.topmenu {
  margin: 0 rem(10px);

  @include s-phone() {
    margin: rem(10px) rem(10px);
  }
}

.link {
  font-family: $font-title;
  font-size: rem(18px);
  margin-right: rem(80px);
  text-transform: uppercase;
  color: $color-link;
  text-decoration: none;

  &:hover {
    color: $color-active-link;
  }

  &:active {
    color: $color-active-link;
  }

  &:last-child {
    margin-right: 0;
  }

  @include tablet() {
    margin-right: rem(35px);
  }

  @include phone() {
    font-size: rem(16px);
    margin-right: rem(20px);
  }

  @include s-phone() {
    margin: rem(10px);
  }

}

.active {
  color: $color-active-link;
  position: relative;

  &:before {
    content: '[ ';
    display: block;
    position: absolute;
    top: -2px;
    left: -7px;
  }
  &:after {
    content: ' ]';
    display: block;
    position: absolute;
    top: -2px;
    right: -7px;
  }
}