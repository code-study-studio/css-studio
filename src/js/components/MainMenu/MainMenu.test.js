import React from 'react';
import MainMenu from './MainMenu';

describe('MainMenu', () => {

  it('should be defined', () => {
    expect(MainMenu).toBeDefined();
  });

  it('should render correctly', () => {
    const wrapper = shallow(<MainMenu />);
    expect(wrapper).toMatchSnapshot();
  });

});
