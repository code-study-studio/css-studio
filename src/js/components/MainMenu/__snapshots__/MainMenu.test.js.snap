// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu should render correctly 1`] = `
<nav>
  <NavLink
    activeClassName="active"
    aria-current="page"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="about"
  >
    About
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/gallery"
  >
    Gallery
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="/works"
  >
    Works
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    to="./contacts"
  >
    Contacts
  </NavLink>
</nav>
`;
